st2actions/st2actions/container/base.py

```diff
@@ -184,6 +193,93 @@ def _do_cancel(self, runner, runnertype_db, action_db, liveaction_db):
 
         return liveaction_db
 
+    def _do_pause(self, runner, runnertype_db, action_db, liveaction_db):
+        try:
+            extra = {'runner': runner}
+            LOG.debug('Performing pause for runner: %s', (runner.runner_id), extra=extra)
+
+            runner.pause()
+
+            liveaction_db = get_liveaction_by_id(liveaction_db.id)
+
+            runner.container_service = None
+        except:
+            _, ex, tb = sys.exc_info()
+            # include the error message and traceback to try and provide some hints.
+            result = {'error': str(ex), 'traceback': ''.join(traceback.format_tb(tb, 20))}
+            LOG.exception('Failed to pause action %s.' % (liveaction_db.id), extra=result)
+        finally:
+            # Always clean-up the auth_token
+            status = liveaction_db.status
+            self._clean_up_auth_token(runner=runner, status=status)
+
+        return liveaction_db
+
+    def _do_resume(self, runner, runnertype_db, action_db, liveaction_db):
+        updated_liveaction_db = None
+
+        try:
+            extra = {'runner': runner}
+            LOG.debug('Performing resume for runner: %s', (runner.runner_id), extra=extra)
+
+            (status, result, context) = runner.resume()
+
+            try:
+                result = json.loads(result)
+            except:
+                pass
+
+            action_completed = status in action_constants.LIVEACTION_COMPLETED_STATES
```

Comment:
It looks like a lot of code is copied here from "run" method.

Since it looks like the same logic which happens at the end of the run when execution finished should also happen here (https://github.com/StackStorm/st2/blob/0536b072dd3a45c1cd55e827e3b8a93b23e59c85/st2actions/st2actions/container/base.py#L116) we should refactor that in some common method and call it in "run" and this method.